bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3415,6 +3421,8 @@ void ReplicaImp::onTransferringCompleteImp(uint64_t newStateCheckpoint) {
 
   sendToAllOtherReplicas(checkpointMsg);
 
+  setIsCollectingState(false);
```

Comment:
You want to set this as last line in the function.
Previous behavior: isCollecting state called directly to ST would return true, right?We want the same behavior so we set to false as last line.
If you put this line here, you change behavior since tryToEnterView execute when collecting state is false.
I dont have good explenation why we run all this when is collecting state is still set to false, need ot check.
Also, it would be better to put a comment that this one should be last in the function.